WordPress/src/main/java/org/wordpress/android/ui/stories/SaveStoryGutenbergBlockUseCase.kt

```diff
@@ -42,63 +49,143 @@ class SaveStoryGutenbergBlockUseCase @Inject constructor(
+    fun buildMediaFileDataWithTemporaryId(mediaFile: MediaFile, temporaryId: String): StoryMediaFileData {
+        return StoryMediaFileData(
+                alt = "",
+                id = TEMPORARY_ID_PREFIX + temporaryId, // mediaFile.id,
+                link = StringUtils.notNullStr(mediaFile.fileURL),
+                type = if (mediaFile.isVideo) "video" else "image",
+                mime = StringUtils.notNullStr(mediaFile.mimeType),
+                caption = "",
+                url = StringUtils.notNullStr(mediaFile.fileURL)
         )
-        val gson = Gson()
-        val storyBlockData: StoryBlockData? = gson.fromJson(jsonString, StoryBlockData::class.java)
-
-        val localMediaId = mediaFile.id
-        // now replace matching localMediaId with remoteMediaId in the mediaFileObjects, obtain the URLs and replace
-        storyBlockData?.mediaFiles?.filter { it.id == localMediaId }?.get(0)?.apply {
-            id = mediaFile.mediaId.toInt()
-            link = mediaFile.fileURL
-            url = mediaFile.fileURL
-
-            // look for the slide saved with the local id key (mediaFile.id), and re-convert to mediaId.
-            val localIdKey = mediaFile.id.toInt()
-            val remoteIdKey = mediaFile.mediaId.toLong()
-            val localSiteId = post.localSiteId.toLong()
-            storiesPrefs.getSlideWithLocalId(
-                    localSiteId,
-                    LocalId(localIdKey)
-            )?.let {
-                it.id = mediaFile.mediaId // update the StoryFrameItem id to hold the same value as the remote mediaID
-                storiesPrefs.saveSlideWithRemoteId(
-                        localSiteId,
-                        RemoteId(remoteIdKey), // use the new mediaId as key
-                        it
-                )
-                // now delete the old entry
-                storiesPrefs.deleteSlideWithLocalId(
-                        localSiteId,
-                        LocalId(localIdKey)
-                )
+    }
+
+    fun buildMediaFileDataWithTemporaryIdNoMediaFile(
+        temporaryId: String,
+        url: String,
+        isVideo: Boolean
+    ): StoryMediaFileData {
+        return StoryMediaFileData(
+                alt = "",
+                id = TEMPORARY_ID_PREFIX + temporaryId, // mediaFile.id,
+                link = url,
+                type = if (isVideo) "video" else "image",
+                mime = "",
+                caption = "",
+                url = url
+        )
+    }
+
+    fun cleanTemporaryMediaFilesStructFoundInAnyStoryBlockInPost(editPostRepository: EditPostRepository) {
+        editPostRepository.update { postModel: PostModel ->
+            val gson = Gson()
+            findAllStoryBlocksInPostContentAndPerformOnEachMediaFilesJsonString(
+                    postModel,
+                    object : DoWithMediaFilesListener {
+                        override fun doWithMediaFilesJsonString(content: String, mediaFilesJsonString: String): String {
```

Comment:
What do you think about this suggestion? The type of the Json string is within the method params. LMK :) 
```suggestion
                        override fun processMediaFilesJson(content: String, mediaFilesJsonString: String): String {
```

Reply:
Kept `doWithMediaFilesJson` and stripped the type postfix in 8b9885f6c9